modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/java/com/liferay/dynamic/data/mapping/form/web/internal/portlet/action/AddFormInstanceRecordMVCActionCommand.java

```diff
@@ -134,6 +134,14 @@ protected void doProcessAction(
 
 		String portletId = _portal.getPortletId(actionRequest);
 
+		if (SessionMessages.contains(
+				actionRequest,
+				portletId +
+					SessionMessages.KEY_SUFFIX_HIDE_DEFAULT_SUCCESS_MESSAGE)) {
+
+			SessionMessages.clear(actionRequest);
```

Comment:
Hey, @carolmariaabb. When you clear the `SessionMessages`, does it get rid of the `KEY_SUFFIX_HIDE_DEFAULT_SUCCESS_MESSAGE` as well? I'm asking you this, because we would be removing it to put it back on line 167, if it does.

Reply:
yeah! But the line 167 is not executed when submitting a new entry. It actually goes to `if (Validator.isNotNull(redirectURL)) {`, so we need to remove this key anyway to display the alert on the success page.

Reply:
Can we remove the key inside the `if (Validator.isNotNull(redirectURL))` scope then? What do you think?

Reply:
Another thing that triggered my curiosity... 😅 
Why do we enter the `if` block when adding a new entry if we haven't set the `redirectURL` setting?

Reply:
That’s a good question. I was also expecting to enter the `else` block, but the `ParamUtil.getString(actionRequest, "redirect", ddmFormInstanceSettings.redirectURL());` returns the `actionRequest` redirect string .

Reply:
Ohh... I see. That's misleading. 😅

Reply:
I can't remove the key inside the `if (Validator.isNotNull(redirectURL))` because the `SessionMessages.clear(actionRequest);` must come before the `SessionMessages.add(actionRequest, portletId + "formInstanceRecordAdded");`.

Reply:
Otherwise I'll remove the `formInstanceRecordAdded` too